pkg/apis/serving/v1beta1/trained_model.go

```diff
@@ -10,14 +11,17 @@ import (
 type TrainedModel struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	Spec              TrainedModelSpec   `json:"spec,omitempty"`
-	Status            TrainedModelStatus `json:"status,omitempty"`
+	// +optional
```

Comment:
These aren't optional.

Reply:
Good catch! I removed the optional comment for Spec, but status should be optional isn't it? When a user just creates a TrainedMode CR, it does not have status.

Reply:
The object should default to status: {} when it's created. Something can only be optional if it's a *